### PR TITLE
Fix ADSR step & add noise

### PR DESCRIPTION
**Pull request by KieronJ**
The broken ADSR step is a rather subtle bug, as it sounds the same most of the time. However it is quite apparent when listening to FFVII's cursor.

The noise algorithm is based upon Dr.Hell's documentation.

### CODE REVIEW

File: src/device/spu/noise.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#include <array>
+
+namespace {
+std::array<int8_t, 64> noise_addition = {
+	{1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0,
+	 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0,
+	 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1,
+	 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1}};
+
+std::array<int8_t, 5> noise_halfcycle = {{0, 84, 140, 180, 210}};
```
Comment (JaCzekanski) [inline src/device/spu/noise.cpp]: These values doesn't fit in 8 bit type - int16_t should be used instead.

**Review by JaCzekanski — CHANGES_REQUESTED** [review]
Tested with Chrono Trigger intro - sea sound effects (noise based) are generated properly.

Won't compile on Clang unless using int16_t for noise_halfcycle.

File: src/device/spu/noise.cpp (hunk 1 above)
Comment (KieronJ) [reply src/device/spu/noise.cpp]: Will work with uint8_t too, fixed.

**Review by JaCzekanski — APPROVED** [review]
LGTM

**Pull request merged**